### PR TITLE
Install doxygen-awesome

### PR DESCRIPTION
**Pull request by nahueespinosa**
### Proposed changes

Installs and configures https://jothepro.github.io/doxygen-awesome-css/.
This modern theme updates the visuals of Doxygen without changing its overall layout too much.

#### Type of change

- [ ] 🐛 Bugfix (change which fixes an issue)
- [x] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

### Checklist

- [x] Lint and unit tests (if any) pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] All commmits have been signed for [DCO](https://developercertificate.org/)

## Additional comments

Here is the output of the documentation build step: [docs.zip](https://github.com/Ekumen-OS/beluga/files/11599991/docs.zip)

### CODE REVIEW

**Comment by nahueespinosa** [comment]
## Before:
Screenshot from 2023-05-30 09-37-45

## After:
Screenshot from 2023-05-30 09-37-29

File: .gitmodules [hunk 1]
```diff
@@ -2,3 +2,6 @@
   path = flatland
   url = git@github.com:avidbots/flatland.git
   branch = ros2-humble
+[submodule "beluga/docs/doxygen-awesome-css"]
+  path = beluga/docs/doxygen-awesome-css
+  url = git@github.com:jothepro/doxygen-awesome-css.git
```
Comment (ivanpauno) [inline .gitmodules:7]: I'm not a big fan of adding everything as a submodule.
For build dependencies I think it's better to use something like `vcs`.

In this case, we could pull the dependency in the CMakeLists that build the docs (with ExternalProject_Add or FetchContent).

**Review by ivanpauno — APPROVED** [review]
LGTM!

File: .gitmodules (hunk 1 above)
Comment (nahueespinosa) [reply .gitmodules]: Done in https://github.com/Ekumen-OS/beluga/pull/204/commits/6a765db8bb518d6c6c3db6a07241b7778e4b3b3a.

**Review by ivanpauno — APPROVED** [review]

**Pull request merged**